### PR TITLE
[jit] Add intrinsics for Math.Round/Floor/Cecil () on SSE41.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/intrinsics.c [hunk 1]
```diff
@@ -1161,6 +1161,26 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
+				int xreg = alloc_xreg (cfg);
+				EMIT_NEW_UNALU (cfg, ins, OP_FCONV_TO_R8_X, xreg, args [0]->dreg);
+				EMIT_NEW_UNALU (cfg, ins, OP_SSE41_ROUNDPD, xreg, xreg);
+				ins->inst_c0 = mode;
+				int dreg = alloc_freg (cfg);
+				EMIT_NEW_UNALU (cfg, ins, OP_EXTRACT_R8, dreg, xreg);
+				return ins;
+			}
+		}
+#endif
```
Comment (filipnavara) [inline mono/mini/intrinsics.c]: Where is the corresponding #if for that? You added #define above..

File: mono/mini/intrinsics.c [hunk 2]
```diff
@@ -1161,6 +1161,26 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 		 * all inputs:
 		 * http://everything2.com/?node_id=1051618
 		 */
+#ifdef MONO_ARCH_SIMD_INTRINSICS
+		if (!cfg->compile_aot && (mono_arch_cpu_enumerate_simd_versions () & SIMD_VERSION_SSE41) && fsig->param_count == 1 && fsig->params [0]->type == MONO_TYPE_R8) {
```
Comment (kumpera) [inline mono/mini/intrinsics.c]: This looks wrong, it should be in the amd64 backend.

Comment (vargaz) [reply mono/mini/intrinsics.c]: All other simd code is in simd-intrinsics.c as well.

File: mono/mini/mini-amd64.c [hunk 3]
```diff
@@ -8311,6 +8311,25 @@ mono_arch_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMetho
 			ins->inst_i1 = args [1];
 		}
 #endif
+
```
Comment (jaykrell) [inline mono/mini/mini-amd64.c]: This is never going to be shared with 32bit x86?

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build OS X i386

**Comment by luhenry** [comment]
@monojenkins build Linux x64 FullAOT+LLVM

**Comment by luhenry** [comment]
@monojenkins build OS X i386

**Pull request merged**